src/tests/Interop/DisabledRuntimeMarshalling/DisabledRuntimeMarshallingNative.cpp

```diff
@@ -24,24 +19,24 @@ struct StructWithWCharAndShort
     WCHAR c;
 };
 
-extern "C" DLL_EXPORT bool STDMETHODCALLTYPE CheckStructWithShortAndBool(StructWithShortAndBool str, short s, bool b)
+extern "C" DLL_EXPORT BOOL STDMETHODCALLTYPE CheckStructWithShortAndBool(StructWithShortAndBool str, short s, BYTE b)
```

Comment:
I would expect the return type to be BYTE here.

Do we have a bug in the JIT implementation of the unmanaged calling conventions for small integer return types?

Reply:
It seems like the JIT is seeing an `int` return type in the signature when it is compiling `IL_STUB_PInvoke` for the delegate created for `DisabledRuntimeMarshalling.PInvokeAssemblyMarshallingEnabled.DelegatesFromExternalAssembly.StructWithDefaultNonBlittableFields`, so no normalization is done:
```scala
    [ 4]  50 (0x032) calli 11FFFFFF
In Compiler::impImportCall: opcode is calli, kind=0, callRetType is int, structSize is 0

               [000029] DACXG------                         ▌  STORE_LCL_VAR int    V15 tmp3         
               [000027] --CXG------                         └──▌  CALL ind unman int   
               [000022] ----------- arg0                       ├──▌  LCL_VAR   struct<DisabledRuntimeMarshallingNative+StructWithShortAndBool, 12> V13 tmp1         
               [000018] ----------- arg1                       ├──▌  LCL_VAR   int    V07 loc3         
               [000019] ----------- arg2                       ├──▌  LCL_VAR   int    V08 loc4         
               [000026] ----------- calli tgt                  └──▌  LCL_VAR   long   V14 tmp2         
```

Reply:
> PInvokeAssemblyMarshallingEnabled

It is expected that the return type will be `int` (matching unmanaged `BOOL`) when the marshalling is enabled.

Maybe the problem is that `CheckStructWithShortAndBool` unmanaged method is used for both the marshaling disabled test (where bool needs to be 1 byte) and marshaling enabled test (where bool needs to be 4 bytes by default)?

Reply:
> Maybe the problem is that CheckStructWithShortAndBool unmanaged method is used for both the marshaling disabled test (where bool needs to be 1 byte) and marshaling enabled test (where bool needs to be 4 bytes by default)?

Yes, that is the general root of the problem and I can update that as such. I was trying to avoid rewriting these tests. They are written in an odd way and seem to be more complex than needed.

Reply:
I think we should fix the tests to avoid mismatched unmanaged/managed signatures. The mismatched unmanaged/managed signatures are always a bug farm.

Reply:
Can/should stdint types be used here?

Reply:
`BYTE` is already used in this test and this isn't where I want to start converting these types to stdint.